? .swp
? calendar/gui/.e-cal-model.c.swp
? calendar/gui/.e-calendar-view.c.swp
? help/C/evolution-2.2-C.omf
? help/C/evolution-2.2.xml
? mail/default/zh_CN/Makefile
? mail/default/zh_CN/Makefile.in
? plugins/calendar-weather
? plugins/groupwise-account-setup/Makefile
? plugins/groupwise-account-setup/Makefile.in
? plugins/groupwise-account-setup/org-gnome-gw-account-setup.eplug
? widgets/misc/test-title-bar
Index: configure.in
===================================================================
RCS file: /cvs/gnome/evolution/configure.in,v
retrieving revision 1.755
diff -u -r1.755 configure.in
--- configure.in	23 Dec 2004 07:00:30 -0000	1.755
+++ configure.in	30 Dec 2004 02:50:19 -0000
@@ -1333,7 +1333,7 @@
 AC_ARG_ENABLE(plugins, [  --enable-plugins=[no/all/list]      Enable plugins.],enable_plugins="$enableval",enable_plugins=base)
 
 dnl Add any new plugins here
-plugins_base="calendar-http groupwise-account-setup plugin-manager" 
+plugins_base="calendar-http calendar-weather groupwise-account-setup plugin-manager" 
 plugins_all="bbdb subject-thread save-attachments prefer-plain save-calendar select-one-source copy-tool mail-to-meeting mail-to-task folder-unsubscribe mark-calendar-offline audio-inline mailing-list-actions groupwise-account-setup"
 
 case x"$enable_plugins" in
@@ -1535,6 +1535,7 @@
 mail/importers/Makefile
 plugins/Makefile
 plugins/calendar-http/Makefile
+plugins/calendar-weather/Makefile
 plugins/plugin-manager/Makefile
 plugins/bbdb/Makefile
 plugins/audio-inline/Makefile
Index: calendar/gui/migration.c
===================================================================
RCS file: /cvs/gnome/evolution/calendar/gui/migration.c,v
retrieving revision 1.32
diff -u -r1.32 migration.c
--- calendar/gui/migration.c	7 Oct 2004 21:03:36 -0000	1.32
+++ calendar/gui/migration.c	30 Dec 2004 02:50:20 -0000
@@ -362,6 +362,7 @@
 }
 
 
+#define WEATHER_BASE_URI "weather://"
 #define WEBCAL_BASE_URI "webcal://"
 #define CONTACTS_BASE_URI "contacts://"
 #define BAD_CONTACTS_BASE_URI "contact://"
@@ -393,7 +394,8 @@
 			 ESourceGroup **on_this_computer,
 			 ESource **personal_source,
 			 ESourceGroup **on_the_web,
-			 ESourceGroup **contacts)
+			 ESourceGroup **contacts,
+			 ESourceGroup **weather)
 {
 	GSList *groups;
 	ESourceGroup *group;
@@ -402,6 +404,7 @@
 	*on_this_computer = NULL;
 	*on_the_web = NULL;
 	*contacts = NULL;
+	*weather = NULL;
 	*personal_source = NULL;
 
 	base_uri = g_build_filename (calendar_component_peek_base_directory (component),
@@ -486,7 +489,15 @@
 
 		*contacts = group;
 	}
-	
+
+	if (!*weather) {
+		/* Create the Weather source group */
+		group = e_source_group_new (_("Weather"), WEATHER_BASE_URI);
+		e_source_list_add_group (source_list, group, -1);
+
+		*weather = group;
+	}
+
 	g_free (base_uri_proto);
 	g_free (base_uri);
 }
@@ -698,14 +709,14 @@
 gboolean
 migrate_calendars (CalendarComponent *component, int major, int minor, int revision, GError **err)
 {
-	ESourceGroup *on_this_computer = NULL, *on_the_web = NULL, *contacts = NULL;
+	ESourceGroup *on_this_computer = NULL, *on_the_web = NULL, *contacts = NULL, *weather = NULL;
 	ESource *personal_source = NULL;
 	gboolean retval = FALSE;
 
 	/* we call this unconditionally now - create_groups either
 	   creates the groups/sources or it finds the necessary
 	   groups/sources. */
-	create_calendar_sources (component, calendar_component_peek_source_list (component), &on_this_computer, &personal_source, &on_the_web, &contacts);
+	create_calendar_sources (component, calendar_component_peek_source_list (component), &on_this_computer, &personal_source, &on_the_web, &contacts, &weather);
 
 	if (major == 1) {
 		xmlDocPtr config_doc = NULL;
@@ -841,6 +852,8 @@
 		g_object_unref (on_the_web);
 	if (contacts)
 		g_object_unref (contacts);
+	if (weather)
+		g_object_unref (weather);
 	if (personal_source)
 		g_object_unref (personal_source);
 
